TodoApp/test/TodoApp.Services.Tests/Services/AddItemServiceTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Threading.Tasks;
+using NSubstitute;
+using NUnit.Framework;
+using TodoApp.Contracts;
+using TodoApp.Contracts.Models;
+using TodoApp.Contracts.Services;
+using TodoApp.Contracts.Services.Factories;
+using TodoApp.Services.Services;
+using TodoApp.Services.Tests.Comparers;
+
+namespace TodoApp.Services.Tests.Services
+{
+    [TestFixture]
+    internal class AddItemServiceTests
+    {
+        private IAddItemService _service;
+        private ITimeService _timeService;
+        private IGuidService _guidService;
+        private IItemRepository _repository;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _repository = Substitute.For<IItemRepository>();
+            _timeService = Substitute.For<ITimeService>();
+            _guidService = Substitute.For<IGuidService>();
+
+            _service = new AddItemService(_repository, _timeService, _guidService);
+        }
+
+        [Test]
+        public async Task AddItemAsync_NewItem_ReturnsSavedItem()
+        {
+            const string newItemText = "new item";
+            var currentTime = new DateTime(2018, 2, 5, 15, 0, 24);
+            var guid = new Guid("6548b7f6-d35c-4075-90d5-3a17e101f2c4");
+            _timeService.GetCurrentDateTime().Returns(currentTime);
+            _guidService.GenerateGuid().Returns(guid);
+            var expectedItem = new Item
+            {
+                Id = guid,
+                Text = newItemText,
+                CreatedAt = currentTime,
+                ModifiedAt = currentTime
+            };
+            var newItem = new Item {Text = newItemText};
+            _repository.AddAsync(Arg.Is<Item>(value
```

Comment:
Create a new extension method for your `ItemEqualityComparer` that will cover this situation. 

Note: Even if you do not compare all the attributes, you might want to create a special comparer - e.g. `ItemIdentifierEqualityComparer` and something similar. But that is not always necessary and when you go that way once, you then have to create new comparers every time (making it a "coding practice") – so, when comparing the object on a subset of its properties, especially with boolean-returning lambda, go with a named function first and a comparer later.
 When comparing on all properties, always use the comparer (since adding properties to the object will require only since change in the comparer to keep the tests up to date)…